google-ads-migration-examples/src/main/java/com.google.ads.googleads.migration/createcampaign/CreateCompleteCampaignBothApisPhase1.java

```diff
@@ -0,0 +1,515 @@
+   */
+  private Campaign createCampaign(AdWordsServicesInterface adWordsServices,
+                                  AdWordsSession session, CampaignBudget budget)
+    throws RemoteException {
+    // Get the CampaignService.
+    CampaignServiceInterface campaignService =
+      adWordsServices.get(session, CampaignServiceInterface.class);
+
+    // Create the campaign.
+    Campaign campaign = new Campaign();
+    campaign.setName("Interplanetary Cruise #" + System.currentTimeMillis());
+
+    // Recommendation: Set the campaign to PAUSED when creating it to prevent
+    // the ads from immediately serving. Set to ENABLED once you've added
+    // targeting and the ads are ready to serve.
+    campaign.setStatus(CampaignStatus.PAUSED);
+
+    BiddingStrategyConfiguration biddingStrategyConfiguration = new BiddingStrategyConfiguration();
+    biddingStrategyConfiguration.setBiddingStrategyType(BiddingStrategyType.MANUAL_CPC);
+    campaign.setBiddingStrategyConfiguration(biddingStrategyConfiguration);
+
+    // You can optionally provide these field(s).
+    campaign.setStartDate(new DateTime().plusDays(1).toString("yyyyMMdd"));
+    campaign.setEndDate(new DateTime().plusDays(30).toString("yyyyMMdd"));
+
+    // Only the budgetId should be sent, all other fields will be ignored by CampaignService.
+    Budget newBudget = new Budget();
+    newBudget.setBudgetId(budget.getId().getValue());
+    campaign.setBudget(newBudget);
+
+    campaign.setAdvertisingChannelType(AdvertisingChannelType.SEARCH);
+
+    // Set the campaign network options to Search and Search Network.
+    NetworkSetting networkSetting = new NetworkSetting();
+    networkSetting.setTargetGoogleSearch(true);
+    networkSetting.setTargetSearchNetwork(true);
+    networkSetting.setTargetContentNetwork(false);
+    networkSetting.setTargetPartnerSearchNetwork(false);
+    campaign.setNetworkSetting(networkSetting);
+
+    // Create the operation.
+    CampaignOperation operation = new CampaignOperation();
+    operation.setOperand(campaign);
+    operation.setOperator(Operator.ADD);
+
+    CampaignOperation[] operations = new CampaignOperation[] {operation};
+
+    // Add the campaign.
+    CampaignReturnValue result = campaignService.mutate(operations);
+
+    Campaign campaignResult = result.getValue(0);
+    // Display the campaign.
+    System.out.printf("Campaign with ID '%d' and name %s was created.%n",
+      campaignResult.getId(),
+      campaignResult.getName());
+    return campaignResult;
+  }
+
+  /**
+   * Retrieves the campaign budget.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @param budgetResourceName resource name of the new campaign budget.
+   * @throws GoogleAdsException if an API request failed with one or more service errors.
+   */
+  private CampaignBudget getBudget(GoogleAdsClient googleAdsClient, long customerId,
```

Comment:
@nwbirnie I personally think we should leave it as is with the duplicated code rather than using inheritance because each of these code samples is expected to represent the codebase at a given snapshot in time. It wouldn't be realistic to inherit in that case.